File: data/src/main/java/com/github/gmetal/data/net/response/MoviesNowPlayingResponseEntity.java
package com.github.gmetal.data.net.response;

import com.bluelinelabs.logansquare.annotation.JsonField;
import com.bluelinelabs.logansquare.annotation.JsonObject;
import com.github.gmetal.data.entity.MovieEntity;

import java.util.List;

@JsonObject
public class MoviesNowPlayingResponseEntity extends BasePagedResponseEntity<MovieEntity> {

    @JsonField(name = "results")
    public List<MovieEntity> mResults;

    @Override
    public List<MovieEntity> getResultObjects() {

        return mResults;
    }
}

File: data/src/main/java/com/github/gmetal/data/entity/response/MoviesNowPlayingResponseEntity.kt
package com.github.gmetal.data.entity.response

import com.github.gmetal.data.entity.MovieEntity
import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
class MoviesNowPlayingResponseEntity : BasePagedResponseEntity(), ResponseEntityList<MovieEntity> {

    @SerialName("results")
    override var resultObjects: List<MovieEntity> = emptyList()
}
